feature/security/gnsi/credentialz/tests/hiba_authentication/hiba_authentication_test.go

```diff
@@ -0,0 +1,690 @@
+package hiba_authentication
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username                 = "testuser"
+	password                 = "i$V5^6IhD*tZ#eg1G@v3xdVZrQwj"
+	userPrivateKeyFilename   = "users/testuser"
+	userCertFilename         = "users/testuser-cert.pub"
+	dutPrivateKeyFilename    = "hosts/dut"
+	dutCertFilename          = "hosts/dut-cert.pub"
+	caPublicKeyFilename      = "ca.pub"
+	hostCertificateVersion   = "v1.0"
+	hostCertificateCreatedOn = 1705962293
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func createNativeRole(t testing.TB, dut *ondatra.DUTDevice, role string) {
+	t.Helper()
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
```

Comment:
can we do this in a non vendor-specific way?

Reply:
I'll look into refactoring this function to make it less vendor-specific.
Do you want me to add the `docker pull` command to the README?

Reply:
Removed vendor-specific settings